src/sdl/utils.cpp

```diff
@@ -1603,7 +1603,7 @@ surface blur_alpha_surface(const surface &surf, int depth)
 		depth = max_blur;
 	}
 
-	boost::circular_buffer<Uint32> queue(max_blur);
+	boost::circular_buffer<Uint32> queue(depth*2+1);
```

Comment:
Just wondering, does this mean `max_blur` is no longer the max blur? If so, maybe a name change is in order or something?

Reply:
No, it means the buffer is now properly sized. Previously, for blur depths larger than half the max_blur minus 1, the buffer constantly overflowed and the calculated average was incorrect.

To calculate the average of a pixel with `x` pixels to either side, the algorithm needs to store `x*2 + 1` pixels. As max_blur is defined as `256` and people rarely used blur depths larger than `127`, this problem was never really noticed.